ts/Series/CompareCumulativeComposition.ts

```diff
@@ -0,0 +1,789 @@
+/* *
+ *
+ *  !!!!!!! SOURCE GETS TRANSPILED BY TYPESCRIPT. EDIT TS FILE ONLY. !!!!!!!
+ *
+ * */
+
+'use strict';
+
+/* *
+ *
+ *  Imports
+ *
+ * */
+
+import type DataExtremesObject from '../Core/Series/DataExtremesObject';
+
+import Axis from '../Core/Axis/Axis.js';
+import Point from '../Core/Series/Point.js';
+import Series from '../Core/Series/Series.js';
+
+const {
+    prototype: {
+        tooltipFormatter: pointTooltipFormatter
+    }
+} = Point;
+
+import U from '../Core/Utilities.js';
+const {
+    addEvent,
+    arrayMax,
+    arrayMin,
+    correctFloat,
+    defined,
+    isArray,
+    isNumber,
+    isString,
+    pick
+} = U;
+
+/* *
+ *
+ *  Declarations
+ *
+ * */
+
+declare module '../Core/Axis/AxisLike' {
+    interface AxisLike {
+        setCompare(compare?: string, redraw?: boolean): void;
+        setCumulative(cumulative?: boolean, redraw?: boolean): void;
+        setModifier(
+            mode: 'compare'|'cumulative',
+            modeState?: boolean|null|string,
+            redraw?: boolean
+        ): void;
+    }
+}
+
+declare module '../Core/Series/PointLike' {
+    interface PointLike {
+        change?: number;
+        cumulativeSum?: number;
+    }
+}
+
+declare module '../Core/Series/SeriesLike' {
+    interface SeriesLike {
+        dataModify?: CompareCumulativeComposition.Additions;
+        setCompare(compare?: string|null, redraw?: boolean): void;
+        setCumulative(cumulative?: boolean|null, redraw?: boolean): void;
```

Comment:
Place all composition states and functions inside the `Additions` class, as this already receives the series instance via constructor. It avoids name conflicts with other compositions and future changes. Also the property for the composition additions should reflect the name of the composition: `Series.compareCumulative` or `DataModifyComposition.Additions` - you choice. The same with axis and point.

```ts
interface SeriesLike {
  compareCumulative?: CompareCumulativeComposition.SeriesAdditions
}
interface AxisLike {
  compareCumulative?: CompareCumulativeComposition.AxisAdditions;
}
interface Point {
  compareCumulative?: CompareCumulativeComposition.PointAdditions; // probably an interface for the two values
}

Reply:
All the methods outside the `Additions` class are API methods and should be available for users methinks.

Changed to `DataModifyComposition.Additions` :)